benches/graphemes.rs

```diff
@@ -0,0 +1,109 @@
+#[macro_use]
+extern crate bencher;
+extern crate unicode_segmentation;
+
+use bencher::Bencher;
+use unicode_segmentation::UnicodeSegmentation;
+
+const TEXT_ARABIC: &str = include_str!("texts/arabic.txt");
+const TEXT_ENGLISH: &str = include_str!("texts/english.txt");
+const TEXT_HINDI: &str = include_str!("texts/hindi.txt");
+const TEXT_JAPANESE: &str = include_str!("texts/japanese.txt");
+const TEXT_KOREAN: &str = include_str!("texts/korean.txt");
+const TEXT_MANDARIN: &str = include_str!("texts/mandarin.txt");
+const TEXT_RUSSIAN: &str = include_str!("texts/russian.txt");
+const TEXT_SOURCE_CODE: &str = include_str!("texts/source_code.txt");
+
+fn graphemes_arabic(bench: &mut Bencher) {
+    bench.iter(|| {
+        for g in UnicodeSegmentation::graphemes(TEXT_ARABIC, true) {
+            bencher::black_box(g);
+        }
+    });
+
+    bench.bytes = TEXT_ARABIC.len() as u64;
+}
+
+fn graphemes_english(bench: &mut Bencher) {
+    bench.iter(|| {
+        for g in UnicodeSegmentation::graphemes(TEXT_ENGLISH, true) {
```

Comment:
the text itself should also pass through `black_box`. Probably doesn't matter given how large it is, but worth a shot.

Alternatively, we can load the file dynamically outside of the `iter()` call.

Reply:
Pushed a fix that does this.